File: java/606.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public String tree2str(TreeNode t) {
        if(t == null){
            return "";
        }
        String s = t.val + "";
        if(t.left == null && t.right == null){
            return s;
        }
        if(t.left != null){
            s = s + "(" + tree2str(t.left) + ")";
        }else{
            s = s + "()";
        }
        if(t.right != null){
            s = s + "(" + tree2str(t.right) + ")";
        }
        return s;
    }
}

File: go/606.go
import "strconv"

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func tree2str(t *TreeNode) string {
	if t == nil {
		return ""
	} else if t.Left == nil && t.Right == nil {
		return strconv.Itoa(t.Val)
	} else if t.Right == nil {
		return strconv.Itoa(t.Val) + "(" + tree2str(t.Left) + ")"
	} else {
		return strconv.Itoa(t.Val) + "(" + tree2str(t.Left) + ")" + "(" + tree2str(t.Right) + ")"
	}
}